gradoop-examples/src/main/java/org/gradoop/examples/aggregation/AggregationExample.java

```diff
@@ -0,0 +1,137 @@
+   */
+  private static final String LABEL_PERSON = "Person";
+
+  /**
+   * Runs the program on the example data graph.
+   *
+   * The example provides an overview over the usage of the aggregate() method.
+   * It both showcases the application of aggregation functions that are already provided by
+   * Gradoop, as well as the definition of custom ones.
+   * Documentation for all available aggregation functions as well as a detailed description of the
+   * aggregate method can be found in the projects wiki.
+   *
+   * @see <a href="https://github.com/dbs-leipzig/gradoop/wiki/Unary-Logical-Graph-Operators">
+   * Gradoop Wiki</a>
+   *
+   * Using the social network graph in the resources directory, the program will:
+   * 1. extract a subgraph only containing vertices which are labeled "person"
+   * 2. concatenate the values of the vertex property "name" of each vertex in the subgraph
+   * 3. count the amount of vertices in the subgraph (aka the amount of persons)
+   * 4. sum up the values of the vertex property "birthday"
+   * 5. add the property "meanAge" to the graph head using a graph head transformation
+   *
+   * @param args arguments
+   */
+  public static void main(String[] args) throws Exception {
+    // init execution environment
+    ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+
+    // create loader
+    FlinkAsciiGraphLoader loader = new FlinkAsciiGraphLoader(GradoopFlinkConfig.createConfig(env));
+
+    // load data
+    loader.initDatabaseFromFile(EXAMPLE_DATA_FILE);
+
+    // get LogicalGraph representation of the social network graph
+    LogicalGraph networkGraph = loader.getDatabase().getDatabaseGraph();
+
+    // execute aggregations and graph-head transformations
+    LogicalGraph result = networkGraph
+      // extract subgraph with edges and vertices which are of interest
+      .vertexInducedSubgraph(new ByLabel<>(LABEL_PERSON))
+      // apply custom VertexAggregateFunctions
+      .aggregate(new VertexAggregateFunction() {
+        @Override
+        public PropertyValue getVertexIncrement(Vertex vertex) {
+          return PropertyValue.create(vertex.getPropertyValue(PROPERTY_KEY_NAME).toString());
+        }
+
+        @Override
+        public PropertyValue aggregate(PropertyValue aggregate, PropertyValue increment) {
+          aggregate.setString(aggregate.getString() + "," + increment.getString());
+          return aggregate;
+        }
+
+        @Override
+        public String getAggregatePropertyKey() {
+          return PROPERTY_KEY_LIST_OF_NAMES;
+        }
+      })
+      // aggregate sum of vertices in order to obtain total amount of persons
+      .aggregate(new VertexCount())
+      // sum up values of the vertex property "birthday"
+      .aggregate(new SumVertexProperty(PROPERTY_KEY_BIRTHDAY))
+      // add computed property "meanAge" to the graph head
+      .transformGraphHead((TransformationFunction<GraphHead>) (current, transformed) -> {
```

Comment:
same goes for a TransformationFunction `AddPropertiesToGraphHead implements TransformationsFunction<GraphHead>`.